src/models/User/createUser/index.js

```diff
@@ -68,7 +69,10 @@ const createUser = async ({
     .insert({ email: normalizedEmail, name: trimmedName, passwordHash })
     .returning('*')
 
-  return { user: savedUser[0], errors: [] }
+  const user = savedUser[0]
```

Comment:
Just out of curiosity, what if insert fails ? Will we catch a proper error that we can handle gracefully or is this whole thing going to throw a server error we can't deal with and break the client ?

Reply:
I think… we should assume it’ll work. 😄

Reply:
Maybe we should wrap it in a `try catch`…
I’m not sure.

```js
 try {
  const savedUser: Array<User> = await database('users')
    .insert({ email: normalizedEmail, name: trimmedName, passwordHash })
    .returning('*')

  const user = savedUser[0]
  UserModel.sendAccountValidationEmail(user)

  return { user, errors: [] }
} catch (error) {
  return { errors: [error] }
}
```

The error won’t have a `field` and maybe not even a `message`, though. I guess we should look into it

Reply:
Okay it won't prevent merging (it's already like this and it's not the point of the PR), but we should plan to find a way to handle external services failures (db, sendgrid, etc..) gracefully so we can have a better errors flow.

Reply:
We could build our own error?

`{ field: generic, message: 'Oops, an unexpected error happened. Please try again.' }`

Reply:
And we should log the actual error, too.

Reply:
I think we should also add a `status: 'SUCCESS'` and `status: 'FAILURE'` since our API will almost should always return a 200.

Reply:
Maybe we should discuss this in an issue in order to spec our error handling flow better. 
I don't think the model layer should be aware of how it's going to be exposed (GraphQL is not our only API). 

We should have model-specific errors that are handled and formatted by the different exposition layers later